optuna/distributions.py

```diff
@@ -295,6 +295,50 @@ def _contains(self, param_value_in_internal_repr):
         return self.low <= value <= self.high
 
 
+class IntLogUniformDistribution(IntUniformDistribution):
+    """A uniform distribution on integers in the log domain.
+
+    This object is instantiated by :func:`~optuna.trial.Trial.suggest_int`, and passed to
+    :mod:`~optuna.samplers` in general.
+
+    Attributes:
+        low:
+            Lower endpoint of the range of the distribution. ``low`` is included in the range.
+        high:
+            Upper endpoint of the range of the distribution. ``high`` is included in the range.
+        step:
+            A step for spacing between values.
+    """
+
+    def __init__(self, low, high, step=1):
+        # type: (int, int, int) -> None
+
+        super(IntLogUniformDistribution, self).__init__(low, high, step)
+
+        if low <= 0.0:
+            raise ValueError(
+                "The `low` value must be larger than 0 for a log distribution "
+                "(low={}, high={}).".format(low, high)
+            )
+
+        if (high - low) % step != 0:
+            raise ValueError(
+                "The `high` - `low` must be a multiple of `step`, but "
+                "low={}, high={}, step={}.".format(low, high, step)
+            )
```

Comment:
This part comes from [automl's ConfigSpace](https://github.com/automl/ConfigSpace/blob/d36a6b275bdffb1382cc0820c602668d66396e5b/ConfigSpace/hyperparameters.pyx#L818-L822) that causes the same error. But it may be removed for consistency to `DiscreteUniformDistribution` in Optuna.

Reply:
> But it may be removed for consistency to DiscreteUniformDistribution in Optuna.

I agree with you. IMO, `IntLogUniformDistribution` can also changes `high` if `(high - low) % step != 0`.
Do you work on it within this PR or create new one?

Reply:
Hi, Thank you for your comment! I work on it within this PR.

Reply:
Done